Venmo/src/main/java/com/braintreepayments/api/venmo/VenmoClient.kt

```diff
@@ -0,0 +1,352 @@
+package com.braintreepayments.api.venmo
+
+import android.content.Context
+import android.net.Uri
+import android.text.TextUtils
+import android.util.Base64
+import androidx.annotation.VisibleForTesting
+import com.braintreepayments.api.BrowserSwitchFinalResult
+import com.braintreepayments.api.BrowserSwitchOptions
+import com.braintreepayments.api.core.AnalyticsEventParams
+import com.braintreepayments.api.core.ApiClient
+import com.braintreepayments.api.core.AppSwitchNotAvailableException
+import com.braintreepayments.api.core.Authorization
+import com.braintreepayments.api.core.BraintreeClient
+import com.braintreepayments.api.core.BraintreeException
+import com.braintreepayments.api.core.BraintreeRequestCodes
+import com.braintreepayments.api.core.ClientToken
+import com.braintreepayments.api.core.Configuration
+import com.braintreepayments.api.core.MetadataBuilder
+import org.json.JSONException
+import org.json.JSONObject
+import java.util.Objects
+
+/**
+ * Used to create and tokenize Venmo accounts. For more information see the [documentation](https://developer.paypal.com/braintree/docs/guides/venmo/overview)
+ */
+class VenmoClient @VisibleForTesting internal constructor(
+    private val braintreeClient: BraintreeClient,
+    private val venmoApi: VenmoApi,
+    private val sharedPrefsWriter: VenmoSharedPrefsWriter
+) {
+    /**
+     * Used for linking events from the client to server side request
+     * In the Venmo flow this will be a Payment Context ID
+     */
+    private var payPalContextId: String? = null
+
+    /**
+     * True if `tokenize()` was called with a Vault request object type
+     */
+    private var isVaultRequest = false
+
+    private constructor(braintreeClient: BraintreeClient, apiClient: ApiClient) : this(
+        braintreeClient, VenmoApi(braintreeClient, apiClient),
+        VenmoSharedPrefsWriter()
+    )
+
+    /**
+     * Initializes a new [VenmoClient] instance
+     *
+     * @param context an Android Context
+     * @param authorization a Tokenization Key or Client Token used to authenticate
+     * @param returnUrlScheme a custom return url to use for browser and app switching
+     */
+    constructor(
+        context: Context,
+        authorization: String,
+        returnUrlScheme: String?
+    ) : this(BraintreeClient(context, authorization, returnUrlScheme))
+
+    @VisibleForTesting
+    internal constructor(braintreeClient: BraintreeClient) : this(
+        braintreeClient,
+        ApiClient(braintreeClient)
+    )
```

Comment:
We should be able to remove this constructor as well with adding default values to the primary constructor.